buildscripts/common.props

```diff
@@ -1,21 +1,35 @@
 <Project>
-  <PropertyGroup>
-    <VersionPrefix>5.0.0</VersionPrefix>
-    <NoWarn>$(NoWarn);CS1591</NoWarn>
-    <RepositoryType>git</RepositoryType>
-    <RepositoryUrl>https://github.com/castleproject/Core</RepositoryUrl>
-  </PropertyGroup>
+	<PropertyGroup>
+		<NoWarn>$(NoWarn);CS1591;CS3014;CS3003;CS3001;CS3021</NoWarn>
+		<RepositoryType>git</RepositoryType>
+		<RepositoryUrl>https://github.com/castleproject/Core</RepositoryUrl>
+		<BuildVersion Condition="'$(APPVEYOR_BUILD_VERSION)'==''">4.0.0</BuildVersion>
+		<BuildVersion Condition="'$(APPVEYOR_BUILD_VERSION)'!=''">$(APPVEYOR_BUILD_VERSION)</BuildVersion>
```

Comment:
> I need a bit more info here. Travis injects build numbers using the TRAVIS_BUILD_NUMBER environment variable whilst AppVeyor uses APPVEYOR_BUILD_VERSION for the full number or APPVEYOR_BUILD_NUMBER. What did you mean when you said something more generic? Version.props?

I don't really have a preference for implementation, but yes it was that fact that this was tied to one CI platform's environment variables. I don't think I realised this was in common.props before, I'd leave it for now since this is the only place it is defined.